src/main/kotlin/com/openlattice/shuttle/Shuttle.kt

```diff
@@ -161,36 +220,43 @@ class Shuttle(
             rowColsToPrint: List<String>,
             sw: Stopwatch
     ) {
-        logger.info("There are ${remaining.incrementAndGet()} batches in process for upload.")
+        val batchUpdate = "There are ${remaining.incrementAndGet()} batches in process for upload."
+        writeLog(flight.name, batchUpdate, OffsetDateTime.now(), IntegrationStatus.IN_PROGRESS)
         val batchSw = Stopwatch.createStarted()
         val batch = try {
             rows.add(chunk.size.toLong())
             val batchCtr = batchCounter.incrementAndGet()
             minRows[batchCtr] = chunk[0]
             impulse(flight, chunk, batchCtr)
         } catch (ex: Exception) {
+            val errorInfo = if (ex is RhizomeRetrofitCallException) {
+                "Server returned ${ex.code} with body: ${ex.body}."
+            } else {
+                "Something went wrong during client side processing. "
+            }
+            writeLog(flight.name, errorInfo, OffsetDateTime.now(), IntegrationStatus.FAILED)
             MissionControl.fail(1, flight, ex, listOf(uploadingExecutor))
         } catch (err: OutOfMemoryError) {
+            writeLog(flight.name, "out of memory error", OffsetDateTime.now(), IntegrationStatus.FAILED)
```

Comment:
we may want to log the actual stacktraces when these errors are caught

Reply:
Agreed most of the time, but OOME is a special case -- lots of things are going to be spontaneously failing, including possibly writing out log files, so keeping it simple is better.

We should probably actually semi-orderly shutdown and restart if we encounter an OOME.